src/assemblyItemsBrowser.js

```diff
@@ -170,51 +159,43 @@ module.exports = React.createClass({
 
 	updateState: function(props, vmTraceIndex)
 	{
-		if (!props.vmTrace)
+		if (!props.vmTrace || !props.vmTrace[vmTraceIndex])
 			return
-		var previousState = this.state.currentSelected
+		var previousIndex = this.state.currentSelected
 		var stateChanges = {}
-
-		var currentAddress = this.state.currentAddress
-		if (!currentAddress)
-			currentAddress = props.vmTrace[vmTraceIndex].address
-		if (props.vmTrace[vmTraceIndex].address && props.vmTrace[vmTraceIndex].address !== this.state.currentAddress)
-		{
-			this.resolveAddress(props.vmTrace[vmTraceIndex].address)
-			stateChanges["currentAddress"] = props.vmTrace[vmTraceIndex].address
-		}
-
-		if (props.vmTrace[vmTraceIndex].stack)
+		
+		if (props.vmTrace[vmTraceIndex].stack) // there's always a stack
 		{
 			var stack = props.vmTrace[vmTraceIndex].stack
 			stack.reverse()
 			stateChanges["currentStack"] = stack
 		}
-
-		var callStackIndex = vmTraceIndex
-		if (vmTraceIndex < previousState)
-			callStackIndex = this.retrieveLastSeenProperty(vmTraceIndex, "depth", props.vmTrace)
-		if (this.state.callStack[callStackIndex] || callStackIndex === 0)
-			stateChanges["currentCallStack"] = this.state.callStack[callStackIndex]
-
-		var storageIndex = vmTraceIndex
-		if (vmTraceIndex < previousState)
-			storageIndex = this.retrieveLastSeenProperty(vmTraceIndex, "storage", props.vmTrace)
-		if (props.vmTrace[storageIndex].storage || storageIndex === 0)
-			stateChanges["currentStorage"] = props.vmTrace[storageIndex].storage
-
-		var memoryIndex = vmTraceIndex
-		if (vmTraceIndex < previousState)
-			memoryIndex = this.retrieveLastSeenProperty(vmTraceIndex, "memory", props.vmTrace)	
-		if (props.vmTrace[memoryIndex].memory || memoryIndex === 0)
-			stateChanges["currentMemory"] = this.formatMemory(props.vmTrace[memoryIndex].memory, 16)
-
-		var callDataIndex = vmTraceIndex
-		if (vmTraceIndex < previousState)
-			callDataIndex = this.retrieveLastSeenProperty(vmTraceIndex, "calldata", props.vmTrace)
-		if (props.vmTrace[vmTraceIndex].calldata || callDataIndex === 0)
-			stateChanges["currentCallData"] = [props.vmTrace[callDataIndex].calldata]
-
+		
+		var currentAddress = this.state.currentAddress
+		var addressIndex = this.shouldUpdateStateProperty("address", vmTraceIndex, previousIndex, props.vmTrace)
+		if (addressIndex > -1)
+		{
+			currentAddress = props.vmTrace[addressIndex].address
+			this.resolveAddress(currentAddress)
+			Object.assign(stateChanges, { "currentAddress": currentAddress })
+		}
+		
+		var depthIndex = this.shouldUpdateStateProperty("depth", vmTraceIndex, previousIndex, props.vmTrace)
```

Comment:
this function returns the index where to get the content of the last property (if needed)
but we use the returned index differently depending on the property...